src/librustdoc/html/static/main.js

```diff
@@ -2562,14 +2560,12 @@ function defocusSearchBar() {
         }
         onEachLazy(document.getElementsByClassName("impl-items"), function(e) {
             onEachLazy(e.getElementsByClassName("associatedconstant"), func);
-            var hiddenElems = e.getElementsByClassName("hidden");
-            var needToggle = false;
-
-            var needToggle = onEachLazy(e.getElementsByClassName("hidden"), function(hiddenElem) {
-                if (hasClass(hiddenElem, "content") === false &&
-                    hasClass(hiddenElem, "docblock") === false) {
-                    return true;
-                }
+            // We transform the DOM iterator into a vec of DOM elements to prevent performance
+            // issues on webkit browsers.
+            var hiddenElems = Array.prototype.slice.call(e.getElementsByClassName("hidden"));
```

Comment:
I'm by no means a JS expert, so this may be totally wrong, but can't you do:

```suggestion
            var hiddenElems = e.getElementsByClassName("hidden").slice();
```

Reply:
No, because, even if it looks like an `Array`, it's not an `Array` but a `HTMLCollection` (more info [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection)). Which is why I called it this way. :)

But thanks for the suggestion!

Reply:
Ah, okay. Why can't you do:

```suggestion
            var hiddenElems = Array.prototype.slice(e.getElementsByClassName("hidden"));
```

though?

Reply:
Explanations [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#array-like_objects). :)

Reply:
I see. JS is so weird :)

It does seem you can do `[].slice.call(e.getElementsByClassName("hidden"))` but I think the way you have it now is probably the clearest.

Thanks for explaining the reasoning to me :)

Reply:
This is a very dark world. I wrote [this blog post](https://blog.guillaume-gomez.fr/articles/2021-01-22+Performance+improvement+on+front-end+generated+by+rustdoc) a few days ago about the things I did for front-end performances. Might be interesting to read. ;)